backend/app/services/blockchain_fetcher.py

```
@@ -622,32 +621,29 @@ def fetch_transactions(
+                # Cache the result
+                self._cache_set(cache_key, result, self.TRANSACTION_CACHE_TTL)
 
-            except Exception as e:
-                logger.warning(f"Failed to fetch transactions using {api_name} API: {e}")
-                continue
+                return result
+            else:
+                # API returned no transactions
+                error_msg = f"No transactions returned from blockchain.info API for wallet {wallet_address}"
+                logger.error(error_msg)
+                raise BlockchainFetchError(error_msg)
```

Ruff (0.14.0)

Abstract `raise` to an inner function

(TRY301)